### PR TITLE
Add various regular expressions in deroffer (g_re_*)

### PR DESCRIPTION
**Pull request by necaris**

### CODE REVIEW

File: src/deroff.rs [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
 // """ Deroff.py, ported to Python from the venerable deroff.c """
+extern crate regex;
+use regex::Regex;
 
 // class Deroffer:
-struct Deroffer {}
+struct Deroffer {
+    g_re_word: &'static Regex,
+    g_re_number: &'static Regex,
+    g_re_esc_char: &'static Regex,
```
Comment (scooter-dangle) [inline src/deroff.rs]: Doesn't look like `g_re_esc_char` is used anywhere. Can remove?

File: src/deroff.rs [hunk 2]
```diff
@@ -1,10 +1,40 @@
 /// A translation of https://github.com/fish-shell/fish-shell/blob/e7bfd1d71ca54df726a4f1ea14bd6b0957b75752/share/tools/deroff.py
 // """ Deroff.py, ported to Python from the venerable deroff.c """
+extern crate regex;
+use regex::Regex;
 
 // class Deroffer:
-struct Deroffer {}
+struct Deroffer {
```
Comment (scooter-dangle) [inline src/deroff.rs]: Instead of a struct for these regexes, could we just make all of them global statics?

File: src/deroff.rs [hunk 3]
```diff
@@ -1,10 +1,40 @@
...
+                r##"(?x)([a-zA-Z_]) |  # word
+                    ([+-]?\d+)      |  # number
+                    \\                 # backslash for escape seqs
+                    "##
+            ),
+            // sequence of not backslash or whitespace
+            g_re_not_backslash_or_whitespace: crate::regex!(r##"[^ \t\n\r\f\v\\]+"##),
+            g_re_newline_collapse: crate::regex!(r##"\n{3,}"##),
+            g_re_font: crate::regex!(
+                r##"\\f(         # Starts with backslash f
```
Comment (scooter-dangle) [inline src/deroff.rs]: Regex should start with `(?x)`.

File: src/util.rs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+extern crate regex;
```
Comment (scooter-dangle) [inline src/util.rs]: Don't need to declare crate when using 2018 edition.

File: src/deroff.rs [hunk 5]
```diff
@@ -1,10 +1,40 @@
 /// A translation of https://github.com/fish-shell/fish-shell/blob/e7bfd1d71ca54df726a4f1ea14bd6b0957b75752/share/tools/deroff.py
 // """ Deroff.py, ported to Python from the venerable deroff.c """
+extern crate regex;
```
Comment (scooter-dangle) [inline src/deroff.rs]: Don't need to declare crate when using 2018 edition.

File: src/deroff.rs (hunk 1 above)
Comment (necaris) [reply src/deroff.rs]: Done in 5e03a24 :-)

File: src/deroff.rs (hunk 3 above)
Comment (necaris) [reply src/deroff.rs]: Done in 5e03a24 :-)

File: src/util.rs (hunk 4 above)
Comment (necaris) [reply src/util.rs]: Done in 5e03a24 :-)

File: src/deroff.rs (hunk 5 above)
Comment (necaris) [reply src/deroff.rs]: Done in 5e03a24 :-)

File: src/deroff.rs (hunk 2 above)
Comment (necaris) [reply src/deroff.rs]: In the interest of maintaining parity with the Python implementation, perhaps we could leave as-is right now? I have a sense that some of the regexes may turn out not to be necessary down the road...

**Pull request merged**